components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+import React, { Component } from 'react';
+import PropsTypes from 'prop-types';
+
+export default class LegendItem extends Component {
+
+    constructor(props) {
+        super(props);
+        this._onMouseClick = this._onMouseClick.bind(this);
+        this._onMouseOver = this._onMouseOver.bind(this);
+        this.onMouseOver = this._onMouseOver.bind(this);
+    }
+
+    componentDidMount() {
+    }
+
+    /**
+     * Handle Mouse click events
+     * @param {*} evt Mouse Click event
+     */
+    _onMouseClick(evt) {
+        return this.props.onClick && this.props.onClick(evt);
+    }
+
+    _onMouseOver(evt) {
+        return this.props.onMouseOver && this.props.onMouseOver(evt);
+    }
+
+    _onMouseOut(evt) {
+        return this.props.onMouseOut && this.props.onMouseOut(evt);
+    }
+
+
+    render() {
+        return (
+            <div
+                onClick={this.props.onClick}
+                onMouseOver={this.props.onMouseOver}
+                onMouseOut={this.props.onMouseOut}
+            >
+                <span
+                    
```

Comment:
remove unnecessary spaces and lines and format

Reply:
Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919

Reply:
Move to separate repo.